pkg/scheduler/state/state.go

```diff
@@ -209,22 +200,20 @@ func (s *stateBuilder) State(reserved map[types.NamespacedName]map[string]int32)
 	zoneMap := make(map[string]struct{})
 	for i := 0; i < len(nodes); i++ {
 		node := nodes[i]
-		if node.Spec.Unschedulable {
-			continue //ignore node that is currently unschedulable
-		}
 
-		taints := node.Spec.Taints
-		if contains(taints, noexec) || contains(taints, nosched) {
-			continue //ignore node that is currently unschedulable
+		if isNodeUnschedulable(node) {
```

Comment:
I'll check what the Kubernetes core scheduler does, I'll report back my findings. :monocle_face:

Reply:
So, for Kubernetes this is configurable since we don't want to change the APIs to make it configurable like Kube, can we use only `no schedule` and `no exec` to get the list of potential nodes and keep the fallback to a single `unknown` zone when there is no zone label on those nodes.

Reply:
like this:
```
func isNodeUnschedulable(node *v1.Node) bool {
	noExec := &v1.Taint{
		Key:    "node.kubernetes.io/unreachable",
		Effect: v1.TaintEffectNoExecute,
	}

	noSched := &v1.Taint{
		Key:    "node.kubernetes.io/unreachable",
		Effect: v1.TaintEffectNoSchedule,
	}

	return node.Spec.Unschedulable ||
		contains(node.Spec.Taints, noExec) ||
		contains(node.Spec.Taints, noSched)
}
```

Reply:
ok - thanks for the update from K8 @pierDipi !!

Let's go ahead with that then and remove control/worker node checks. The net change in this file will be the addition of a new zone called `Unknown` for those nodes that are missing the "topology.kubernetes.io/zone" label. Sounds good.

Will these changes be done in this PR? or you'd like me to do it separately?